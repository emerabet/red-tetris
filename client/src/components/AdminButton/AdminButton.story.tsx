import React from 'react';
import { storiesOf } from '@storybook/react';
import AdminButton from './AdminButton';

storiesOf('AdminButton', module)
   .add('normal', () => (
      <AdminButton
        text="start"
        play={() => 'void' }
      />
   ));
