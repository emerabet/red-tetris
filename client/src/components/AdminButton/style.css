.shimmer_start {
    text-align: center;
    color: rgba(255, 255, 255, 0.1);
    background: -webkit-gradient(linear, left top, right top, from(#ed1c5b), to(#ed1c5b), color-stop(0.5, #fff));
    background: -moz-gradient(linear, left top, right top, from(#ed1c5b), to(#ed1c5b), color-stop(0.5, #fff));
    background: gradient(linear, left top, right top, from(#ed1c5b), to(#ed1c5b), color-stop(0.5, #fff));
    -webkit-background-size: 25px 100%;
    -moz-background-size: 25px 100%;
    background-size: 25px 100%;
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    -webkit-animation-name: shimmer;
    -moz-animation-name: shimmer;
    animation-name: shimmer;
    -webkit-animation-duration: 2s;
    -moz-animation-duration: 2s;
    animation-duration: 2s;
    -webkit-animation-iteration-count: infinite;
    -moz-animation-iteration-count: infinite;
    animation-iteration-count: infinite;
    background-repeat: no-repeat;
    background-position: 0 0;
    background-color: #ed1c5b;
  }
  .shimmer_restart {
    text-align: center;
    color: rgba(255, 255, 255, 0.1);
    background: -webkit-gradient(linear, left top, right top, from(#f8b500), to(#f8b500), color-stop(0.5, #fff));
    background: -moz-gradient(linear, left top, right top, from(#f8b500), to(#f8b500), color-stop(0.5, #fff));
    background: gradient(linear, left top, right top, from(#f8b500), to(#f8b500), color-stop(0.5, #fff));
    -webkit-background-size: 25px 100%;
    -moz-background-size: 25px 100%;
    background-size: 25px 100%;
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    -webkit-animation-name: shimmer;
    -moz-animation-name: shimmer;
    animation-name: shimmer;
    -webkit-animation-duration: 2s;
    -moz-animation-duration: 2s;
    animation-duration: 2s;
    -webkit-animation-iteration-count: infinite;
    -moz-animation-iteration-count: infinite;
    animation-iteration-count: infinite;
    background-repeat: no-repeat;
    background-position: 0 0;
    background-color: #f8b500;
  }
  @-moz-keyframes shimmer {
    0% {
      background-position: top left;
    }
    100% {
      background-position: top right;
    }
  }
  @-webkit-keyframes shimmer {
    0% {
      background-position: top left;
    }
    100% {
      background-position: top right;
    }
  }
  @-o-keyframes shimmer {
    0% {
      background-position: top left;
    }
    100% {
      background-position: top right;
    }
  }
  @keyframes shimmer {
    0% {
      background-position: top left;
    }
    100% {
      background-position: top right;
    }
  }
  
  .button-container{
    color: #eee;
    cursor: pointer;
    overflow: hidden;
    height: 40px;
    width: 125px;
    text-align: center;
    vertical-align: middle;
    line-height: 32px;
    border-style: solid;
    border-width: 3px;
    border-radius: 5px;
    -webkit-transition: 100ms;
    -moz-transition: 100ms;
    -o-transition: 100ms;
    transition: 100ms;
    margin: 10px;
    margin-top: 60px;
  }
  
  .button_start {
    border-color: #ed1c5b;
  }

  .button_restart {
      border-color: #f8b500;
  }

  .button_restart:hover {
    background-color: #ed1c5b;
}

.button_start:hover {
    background-color: #f8b500;
  }

  .button-container:hover {
    -webkit-transition: 100ms;
    -moz-transition: 100ms;
    -o-transition: 100ms;
    transition: 100ms;
  }
  
  .button-text{
    width: 125px;
    font-family: helvetica;
    font-size: 28px;
    font-weight: 100; 
    z-index: 1;
    position: absolute;
  }
  
  
  