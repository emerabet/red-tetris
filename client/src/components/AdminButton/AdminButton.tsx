import React from 'react';

import './style.css';

interface AdminButtonProps {
  text: string;
  play: () => void;
}

const AdminButton: React.SFC<AdminButtonProps> = (props) => {

  return (
    <div className={`button-container button_${props.text}`} onClick={props.play}>
      <div className={`button-text shimmer_${props.text}`}>{props.text}</div>
    </div>

  );
};

export default AdminButton;
