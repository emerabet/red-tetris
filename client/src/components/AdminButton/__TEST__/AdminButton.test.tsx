import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AdminButton from '../AdminButton';

configure({ adapter: new Adapter() });

it('renders correctly with defaults', () => {
  const button = renderer.create(<AdminButton text="start" play={() => {}}/>).toJSON();
  expect(button).toMatchSnapshot();
});

it('Test click event', () => {
  const mockCallBack = jest.fn();

  const button = shallow((<AdminButton text="start" play={mockCallBack} />));
  button.find('.button-container').simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
