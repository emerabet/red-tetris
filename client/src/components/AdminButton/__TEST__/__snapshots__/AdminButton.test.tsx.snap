// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="button-container button_start"
>
  <div
    className="button-text shimmer_start"
  >
    start
  </div>
</div>
`;
